### PR TITLE
Core/Indexer API changes

### PR DESCRIPTION
**Pull request by thibault-martinez**
https://github.com/iotaledger/iota.go/pull/618

### CODE REVIEW

File: sdk/src/client/node_api/core/routes.rs [hunk 1]
```diff
@@ -306,25 +309,25 @@ impl ClientInner {
     }
 
     /// Finds a slot commitment by slot index and returns it as object.
-    /// GET /api/core/v3/commitments/by-index/{index}
+    /// GET /api/core/v3/commitments/by-slot/{slot}
     pub async fn get_slot_commitment_by_index(&self, slot_index: SlotIndex) -> Result<SlotCommitment> {
```
Comment (Thoralf-M) [inline sdk/src/client/node_api/core/routes.rs]: Should we also rename the functions?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Yes

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs]: Done

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/client/node_api/core/routes.rs [hunk 2]
```diff
@@ -86,9 +87,10 @@ impl ClientInner {
     }
 
     /// Checks if the account is ready to issue a block.
-    /// GET /api/core/v3/accounts/{accountId}/congestion
+    /// GET /api/core/v3/accounts/{bech32Address}/congestion
     pub async fn get_account_congestion(&self, account_id: &AccountId) -> Result<CongestionResponse> {
-        let path = &format!("api/core/v3/accounts/{account_id}/congestion");
+        let bech32_address = account_id.to_bech32(self.get_bech32_hrp().await?);
+        let path = &format!("api/core/v3/accounts/{bech32_address}/congestion");
```
Comment (Alex6323) [inline sdk/src/client/node_api/core/routes.rs:93]: hmm ... the method signature is not correct yet on purpose, right?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:93]: What do you mean ?

Comment (Thoralf-M) [reply sdk/src/client/node_api/core/routes.rs:93]: Maybe it should accept a bech32 address?

Comment (thibault-martinez) [reply sdk/src/client/node_api/core/routes.rs:93]: I don't know tbh, I think I prefer the IDs 🤷🏻‍♂️

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**